client/src/modules/list/components/ListHeader.jsx

```diff
@@ -0,0 +1,237 @@
+import React, { Fragment, PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { withRouter } from 'react-router-dom';
+
+import { ListIcon } from 'assets/images/icons';
+import { updateList } from 'modules/list/model/actions';
+import { RouterMatchPropType } from 'common/constants/propTypes';
+import { whiteSpaceOnly } from 'common/utils/helpers';
+
+class ListHeader extends PureComponent {
+  constructor(props) {
+    super(props);
+
+    const { details } = this.props;
+    const { description, name } = details;
+
+    this.state = {
+      description,
+      isDescriptionTextareaVisible: false,
+      isNameInputVisible: false,
+      name
+    };
+
+    this.descriptionTextarea = React.createRef();
+    this.nameInput = React.createRef();
+  }
+
+  componentDidMount() {
+    document.addEventListener('keydown', this.handleKeyPress);
+    document.addEventListener('mousedown', this.handleClick);
+  }
+
+  componentDidUpdate() {
+    this.descriptionTextarea.current &&
+      this.descriptionTextarea.current.focus();
+    this.nameInput.current && this.nameInput.current.focus();
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('keydown', this.handleKeyPress);
+    document.removeEventListener('mousedown', this.handleClick);
+  }
+
+  handleNameInputVisibility = () =>
+    this.setState(({ isNameInputVisible }) => ({
+      isNameInputVisible: !isNameInputVisible
+    }));
+
+  handleDescriptionTextareaVisibility = () =>
+    this.setState(({ isDescriptionTextareaVisible }) => ({
+      isDescriptionTextareaVisible: !isDescriptionTextareaVisible
+    }));
+
+  handleNameChange = event => {
+    const {
+      target: { value }
+    } = event;
+
+    this.setState({ name: value });
+  };
+
+  handleDescriptionChange = event => {
+    const {
+      target: { value }
+    } = event;
+
+    this.setState({ description: value });
+  };
+
+  handleKeyPress = event => {
+    const { isDescriptionTextareaVisible } = this.state;
+    const { code } = event;
+
+    if (code === 'Enter' || code === 'Escape') {
+      const action = isDescriptionTextareaVisible
+        ? this.handleDescriptionUpdate
+        : this.handleNameUpdate;
+
+      action();
+    }
+  };
+
+  handleClick = event => {
```

Comment:
this handler is invoking every time you click and anywhere you click, check the same issue in cohortHeader